backend/spec/controllers/spree/admin/users_controller_spec.rb

```diff
@@ -22,198 +30,193 @@
-      user.spree_roles << Spree::Role.find_or_create_by(name: 'admin')
-      spree_post :index
-      expect(response).to render_template :index
-    end
+    context "with ability to admin users" do
+      stub_authorization! do |_user|
+        can [:manage], Spree.user_class
+      end
 
-    it "allows admins to update a user's API key" do
-      user.spree_roles << Spree::Role.find_or_create_by(name: 'admin')
-      expect(mock_user).to receive(:generate_spree_api_key!).and_return(true)
-      spree_put :generate_api_key, id: mock_user.id
-      expect(response).to redirect_to(spree.edit_admin_user_path(mock_user))
-    end
+      it 'can visit index' do
+        spree_post :index
+        expect(response).to be_success
+      end
 
-    it "allows admins to clear a user's API key" do
-      user.spree_roles << Spree::Role.find_or_create_by(name: 'admin')
-      expect(mock_user).to receive(:clear_spree_api_key!).and_return(true)
-      spree_put :clear_api_key, id: mock_user.id
-      expect(response).to redirect_to(spree.edit_admin_user_path(mock_user))
-    end
+      it "allows admins to update a user's API key" do
+        expect {
+          spree_put :generate_api_key, id: user.id
+        }.to change { user.reload.spree_api_key }
+        expect(response).to redirect_to(spree.edit_admin_user_path(user))
+      end
 
-    it 'deny access to users with an bar role' do
-      user.spree_roles << Spree::Role.find_or_create_by(name: 'bar')
-      Spree::Ability.register_ability(BarAbility)
-      spree_post :index
-      expect(response).to redirect_to '/unauthorized'
+      it "allows admins to clear a user's API key" do
+        user.generate_spree_api_key!
+        expect {
+          spree_put :clear_api_key, id: user.id
+        }.to change{ user.reload.spree_api_key }.to(nil)
+        expect(response).to redirect_to(spree.edit_admin_user_path(user))
+      end
     end
 
-    it 'deny access to users with an bar role' do
-      user.spree_roles << Spree::Role.find_or_create_by(name: 'bar')
-      Spree::Ability.register_ability(BarAbility)
-      spree_post :update, { id: '9' }
-      expect(response).to redirect_to '/unauthorized'
-    end
+    context "without ability to admin users" do
+      stub_authorization! do |_user|
+      end
 
-    it 'deny access to users without an admin role' do
-      allow(user).to receive_messages has_spree_role?: false
-      spree_post :index
-      expect(response).to redirect_to '/unauthorized'
+      it 'denies access' do
+        spree_post :index
+        expect(response).to redirect_to '/unauthorized'
+      end
     end
   end
 
   describe "#create" do
     let(:dummy_role) { Spree::Role.create(name: "dummyrole") }
 
-    before do
-      use_mock_user
-      allow(mock_user).to receive_messages(:spree_roles= => true, :stock_locations= => true)
-      user.spree_roles << Spree::Role.find_or_create_by(name: 'admin')
+    # The created user
+    let(:user) { Spree.user_class.last }
```

Comment:
This is confusing me, even with the comment. It overrides the  `let(:user) { create(:user) }` on line 5?  I thought that user on line 5 was the logged in (admin) user, but here it's the user created by the admin creating users? But only if `user` is first accessed (and thus lazily initialized) _after_ the action to create a user?  What is determining the logged in (admin) user then?

I think it might be less confusing if you either called this one by a different name (`let(:created_user)`) if things will still work if you do so, and/or skip the clever `let` altogether, and just do a `created_user = Spree.user_class.last` in context of the expectations.  If I'm understanding how the tests are actually working, which I may not be.